File: lab1-js/lab_js_oop/circle.js
import {Figure} from './figure.js'
import {Color} from './color.js'

export class Circle extends Figure {
    constructor(radius, color) {
        super()
        this.type = 'Круг'
        this.radius = radius
        this.color = new Color(color)
    }

    area() {
        return Math.PI * this.radius * this.radius
    }

    repr() {
        return `${this.type} ${this.color.color} цвета радиусом ${this.radius} площадью ${this.area()}`
    }
}

File: lab1-test/lab_python_oop/circle.py
from .figure import Figure
from .color import Color
import math


class Circle(Figure):
    type = "Круг"

    def __init__(self, radius, color):
        self.radius = radius
        self.color = Color(color)

    def area(self):
        return math.pi * (self.radius ** 2)

    def __repr__(self):
        return '{} {} цвета радиусом {} площадью {}.'.format(
            self.type,
            self.color.val,
            self.radius,
            self.area()
        )
